### ISSUE TITLE
module export-data : missing internationalization support for some strings

### ISSUE THREAD

**Issue by MathieuDerelle** [highcharts/highcharts#13844]
# Description of the feature

There are 2 strings that are difficult to translate in module export-data ('DateTime' + 'Category' x2) : 

https://github.com/highcharts/highcharts/blob/e5d84dc0abbbc5edaa857870952b99f0227400c2/ts/modules/export-data.src.ts#L447-L454

# Library related to the feature
Highcharts/modules/export-data

# Workaround

At the moment, you need to provide an axis title or a custom `columnHeaderFormatter`

Problems : 

1. without using the deprecated option `xAxis.title.enabled`, you can not set `xAxis.title.text` to `null` (to not display it on chart) and have a title in data export

https://api.highcharts.com/highcharts/xAxis.title.enabled
> enabled: boolean
> Deprecated. Set the text to null to disable the title.
> Defaults to undefined.

2. even with an `xAxis.title.text`, "Category" is still present in the export and can not be translated (I guess it's the string on line 448)

**Comment by pawelfus** [comment]
Hi @MathieuDerelle 

Thank you for reporting the issue. Could you tell us which version of Highcharts you have? 

New options are available since v8.1.0:
- `"Category"` -> https://api.highcharts.com/highcharts/lang.exportData.categoryDatetimeHeader
- `"DateTime"` -> https://api.highcharts.com/highcharts/lang.exportData.categoryHeader

Regarding `"Chart"` -> you are right, we can set default reference to `lang.exportData.tableCaption` (+description that option is used only when `chart.title.text` and `chart.exporting.tableCaption` are not defined). What do you think @oysteinmoseng @karolkolodziej ?

**Comment by MathieuDerelle** [comment]
ah, sorry, the "search in repository" brought me to an old version and I did not see those new options
(I'm using 8.1.2)

**Comment by MathieuDerelle** [comment]
I am doing something wrong ? I still can not make `xAxis.title.text` used as header for the categories in the export  
In a nutshell, I'd like to use an option `categoryHeader` for each chart (not defined globally as `lang.exportData.categoryHeader`)

my workaround at the moment : 
```
    csv: {
      columnHeaderFormatter: (item, key, keyLength) => {
        if (item) {
          return false // use highchart defaults
        } else {
          return xAxisTitle
        }
      }
    },
```

**Comment by MathieuDerelle** [comment]
is it normal that you are not hosting all modules ? (https://code.highcharts.com/)

had to use https://cdnjs.com/libraries/highcharts for module export data

**Comment by pawelfus** [comment]
Let me explain, @MathieuDerelle:

- use [`lang.exportData.categoryHeader`](https://api.highcharts.com/highcharts/lang.exportData.categoryHeader) to translate **default** "Category" text
- use [`exporting.csv.columnHeaderFormatter`](https://api.highcharts.com/highcharts/exporting.csv.columnHeaderFormatter) if you want to define each category name and/or define different categories' headers for different charts

In a nutshell, your "workaround" is a valid solution to the case you have.

PS: We do host all of our modules, see: https://code.highcharts.com/modules/export-data.js - if you find any module missing, let us know.

**Comment by karolkolodziej** [comment]
I agree with what Pawel said.
IMHO, if the hardcoded text "Chart" exists in the code, it should be replaced with the lang reference.
It might be called `lang.exportData.tableCaption`

**Comment by oysteinmoseng** [comment]
Agree as well. We are working on internationalization at the moment, and any remaining hardcoded strings will be converted to lang options as part of this process.

**Comment by pawellysy** [comment]
unstale
